builtin/mainmenu/tab_settings.lua

```diff
@@ -333,6 +359,58 @@ local function handle_settings_buttons(this, fields, tabname, tabdata)
 		ddhandled = true
 	end
 
+	for i = 1, #labels.shadow_levels do
+		if fields["dd_shadows"] == labels.shadow_levels[i] then
+			core.settings:set("shadow_levels", dd_options.shadow_levels[2][i])
+			ddhandled = true
+		end
+	end
+
+	if fields["dd_shadows"] == labels.shadow_levels[1] then
+		core.settings:set("enable_dynamic_shadows", "false")
+		ddhandled = true
+	else
+		core.settings:set("enable_dynamic_shadows", "true")
+		core.settings:set("shadow_strength", "0.35")
+		ddhandled = true
```

Comment:
I think this is causing the lua warning in the CI run (ddhandled is set to true in both branches so the value will always be true). This one should be removed.

Reply:
Done.